src/sema/yul/ast.rs

```diff
@@ -124,7 +125,7 @@ pub struct YulFunction {
     pub name: String,
     pub params: Arc<Vec<Parameter>>,
     pub returns: Arc<Vec<Parameter>>,
-    pub body: Vec<YulStatement>,
+    pub body_block: YulBlock,
```

Comment:
I'm not sure why this was renamed.

I'm guessing here: did you want to avoid `.body.body`? Maybe it's better to rename `YulBlock` member `body` to something else like `statements`

Reply:
Yes, I wanted to avoid `.body.body`. I am not keen on `statements`, because the function's body is indeed a YulBlock. Does anything else come to your mind?

Reply:
Just to be clear, this is what I had in mind (hope this is not redundant).

```rust
pub struct YulBlock {
    pub loc: pt::Loc,
    pub reachable: bool,
    pub next_reachable: bool,
    pub statements: Vec<YulStatement>,
}
```

Reply:
Ah, I misunderstood. I thought you were referring to `struct YulFunction`.